File: java/3101. Count Alternating Subarrays.java
class Solution {
    public long countAlternatingSubarrays(int[] nums) {
        long answer = 0;
        int left = 0;
        for (int i = 0; i < nums.length; ) {
            if (i == left) {
                answer += 1;
                i++;
            } else if (nums[i] != nums[i-1]) {
                answer += i - left + 1;
                i++;
            } else {
                left = i;
            }
        }
        return answer;
    }
}

File: go/3101. Count Alternating Subarrays.go
func countAlternatingSubarrays(nums []int) int64 {
	answer := int64(0)
	left := 0
	for right := 0; right < len(nums); {
		if right == left {
			answer += 1
			right += 1
		} else if nums[right] != nums[right-1] {
			answer += int64(right - left + 1)
			right += 1
		} else {
			left = right
		}
	}
	return answer
}